components/scream/src/physics/p3/p3_functions_update_prognostics_impl.hpp

```diff
@@ -0,0 +1,93 @@
+#ifndef P3_FUNCTIONS_UPDATE_PROGNOSTICS_IMPL_HPP
+#define P3_FUNCTIONS_UPDATE_PROGNOSTICS_IMPL_HPP
+
+#include "p3_functions.hpp"
+#include "p3_constants.hpp"
+
+namespace scream {
+namespace p3 {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::update_prognostic_ice(const Spack& qcheti, const Spack& qccol,
+    const Spack& qcshd,  const Spack& nccol,  const Spack& ncheti, const Spack& ncshdc,
+    const Spack& qrcol,  const Spack& nrcol,  const Spack& qrheti, const Spack& nrheti,
+    const Spack& nrshdr, const Spack& qimlt,  const Spack& nimlt,  const Spack& qisub,
+    const Spack& qidep,  const Spack& qinuc,  const Spack& ninuc,  const Spack& nislf,
+    const Spack& nisub,  const Spack& qiberg, const Spack& exner,  const Spack& xxls,
+    const Spack& xlf,    const bool log_predictNc, const bool log_wetgrowth, const Scalar dt,
+    const Spack& nmltratio, const Spack& rhorime_c, Spack& th, Spack& qv, Spack& qitot,
+    Spack& nitot, Spack& qirim, Spack& birim, Spack& qc,  Spack& nc, Spack& qr,
+    Spack& nr)
+{
+
+qc = qc - (qcheti + qccol + qcshd + qiberg) * dt;
```

Comment:
All the lines in this function need to be indented better. Emacs can auto-indent code blocks if that's what you're using.